### PR TITLE
Add Talend Restlet Framework fingerprint

### PR DESCRIPTION
**Pull request by mkienow-r7**
## Description
Adds [Talend Restlet Framework](https://restlet.talend.com/) HTTP server fingerprint.

## Motivation and Context
Improved coverage

## How Has This Been Tested?
* `bundle exec ./bin/recog_verify xml/http_servers.xml`
* `rake tests`

## Types of changes

- New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have updated the documentation accordingly (or changes are not required).
- [x] I have added tests to cover my changes (or new tests are not required).
- [x] All new and existing tests passed.

### CODE REVIEW

File: xml/http_servers.xml [hunk 1]
```diff
@@ -4828,6 +4828,15 @@
     <param pos="0" name="service.cpe23" value="cpe:/a:ballerina:ballerina:{service.version}"/>
   </fingerprint>
 
+  <fingerprint pattern="^Restlet-Framework/([\d.]+)$">
```
Comment (cbarnard-r7) [inline xml/http_servers.xml:4831]: The following cases pass with the current regex and should likely fail.

`Restlet-Framework/2.4.3.`
`Restlet-Framework/2.467..........3`

I believe this regex resolves the issue: `^Restlet-Framework/(\d+(\.?\d+)+)$`

Version info is captured in group 1.

**Review by cbarnard-r7 — CHANGES_REQUESTED** [review]
Let me know if you think it's worth changing.

File: xml/http_servers.xml (hunk 1 above)
Comment (mkienow-r7) [reply xml/http_servers.xml]: That's a valid point. I've been following the existing `([\d.]+)` pattern spread throughout the databases, but I think a pattern that is more precise to known samples would be good. That allows us to catch deviations from known samples for further investigation.

Comment (mkienow-r7) [reply xml/http_servers.xml]: I've modified the version pattern to `(\d+(?:\.\d+)*)` that allows a single version with no period character and also prevents matching version strings that contain multiple period characters in sequence.

```
$ echo "Restlet-Framework/2" | bundle exec ./bin/recog_match xml/http_servers.xml
MATCH: {"matched"=>"Restlet Framework - REST API framework for Java", "service.vendor"=>"Talend", "service.product"=>"Restlet", "service.version"=>"2", "service.cpe23"=>"cpe:/a:talend:restlet:2", "service.protocol"=>"http", "fingerprint_db"=>"http_header.server", "data"=>"Restlet-Framework/2"}

$ echo "Restlet-Framework/2.4" | bundle exec ./bin/recog_match xml/http_servers.xml
MATCH: {"matched"=>"Restlet Framework - REST API framework for Java", "service.vendor"=>"Talend", "service.product"=>"Restlet", "service.version"=>"2.4", "service.cpe23"=>"cpe:/a:talend:restlet:2.4", "service.protocol"=>"http", "fingerprint_db"=>"http_header.server", "data"=>"Restlet-Framework/2.4"}

$ echo "Restlet-Framework/2.4.3" | bundle exec ./bin/recog_match xml/http_servers.xml
MATCH: {"matched"=>"Restlet Framework - REST API framework for Java", "service.vendor"=>"Talend", "service.product"=>"Restlet", "service.version"=>"2.4.3", "service.cpe23"=>"cpe:/a:talend:restlet:2.4.3", "service.protocol"=>"http", "fingerprint_db"=>"http_header.server", "data"=>"Restlet-Framework/2.4.3"}

$ echo "Restlet-Framework/2.40.....3" | bundle exec ./bin/recog_match xml/http_servers.xml
FAIL: Restlet-Framework/2.40.....3
```

**Review by luciddr34m3r — APPROVED** [review]
LGTM

**Review by luciddr34m3r — APPROVED** [review]

**Review by luciddr34m3r — APPROVED** [review]

**Review by cbarnard-r7 — APPROVED** [review]
LGTM

**Pull request merged**